tests/unit/src/flixel/FlxGTest.hx

```diff
@@ -104,4 +105,80 @@ class FlxGTest extends FlxTest
 	{
 		Assert.areEqual(480, FlxG.height);
 	}
+	
+	@Test
+	function testCenter()
+	{
+		final sprite = new FlxSprite();
+		sprite.makeGraphic(100, 100);
+		sprite.origin.set(100, 100);
+		sprite.offset.set(100, 100);
+		sprite.scale.set(2, 4);
+		sprite.angle = 180;
+		sprite.pixelPerfectPosition = true;
+		final graphicBounds = sprite.getGraphicBounds();
+		final offset = FlxPoint.get(sprite.x - graphicBounds.x, sprite.y - graphicBounds.y);
+		final center = FlxPoint.get((FlxG.width - graphicBounds.width) / 2 + offset.x, (FlxG.height - graphicBounds.height) / 2 + offset.y);
+		final offCenter = center.copyTo().add(1000, 1000);
+		
+		sprite.setPosition(offCenter.x, offCenter.y);
+		FlxG.center(sprite, X);
+		FlxAssert.areNear(sprite.x, center.x);
+		FlxAssert.areNear(sprite.y, offCenter.y);
+		
+		sprite.setPosition(offCenter.x, offCenter.y);
+		FlxG.center(sprite, Y);
+		FlxAssert.areNear(sprite.x, offCenter.x);
+		FlxAssert.areNear(sprite.y, center.y);
+		
+		sprite.setPosition(offCenter.x, offCenter.y);
+		FlxG.center(sprite, XY);
+		FlxAssert.areNear(sprite.x, center.x);
+		FlxAssert.areNear(sprite.y, center.y);
+		
+		sprite.setPosition(offCenter.x, offCenter.y);
+		FlxG.center(sprite);
+		FlxAssert.areNear(sprite.x, center.x);
+		FlxAssert.areNear(sprite.y, center.y);
+		
+		offset.put();
+		graphicBounds.put();
+		offCenter.put();
+		center.put();
+	}
```

Comment:
I also want tests against hard coded values, here, but I also want tests for with and without specific sprite features. What if we make a helper function `assertCenterGraphic(sprite, expectedX, expectedY)` that tests each X, Y, XY and no args.

Alternatively, I think it's okay to test that the various args work as expected with one simple case, and then simply test XY for all future tests of scale, origin, offset, angle, rotation and so on